File: Sesion 7/Actividades/Actividad 1/actividad_1.py
class Cliente:
    def __init__(self, nombre, apellido, correo, direccion, numero_tarjeta):
        self.nombre = nombre
        self.apellido = apellido
        self.correo = correo
        self.direccion = direccion
        self.numero_tarjeta = numero_tarjeta

    def obtener_informacion(self):
        info = ""
        info += "Nombre: " + self.nombre + "\n"
        info += "Apellido: " + self.apellido + "\n"
        info += "Correo: " + self.correo + "\n"
        info += "Direccion: " + self.direccion + "\n"
        info += "Numero de tarjeta: " + self.numero_tarjeta + "\n"
        return info

class ClienteSeguro(Cliente):
    def __init__(self, nombre, apellido, correo, direccion, numero_tarjeta):
        super().__init__(nombre, apellido, correo, direccion, numero_tarjeta)

    def encriptar_informacion(self):
        encriptacion = ""
        for c in self.correo:
            encriptacion += str(ord(c)) # Convierte cada letra en ASCII
        return encriptacion

    def verificar_autenticidad(self, correo_ingresado, numero_tarjeta_ingresado):
        return correo_ingresado == self.correo and numero_tarjeta_ingresado == self.numero_tarjeta

cliente_seguro1 = ClienteSeguro("goku", "sayayin 1000", "gokulegana@hotmail_gmail_twitter.com", "Bajo un Puente", "1234 5678 9012 3456")

print("Informacion del cliente:")
print(cliente_seguro1.obtener_informacion())

print("Informacion encriptada:", cliente_seguro1.encriptar_informacion())

autenticado = cliente_seguro1.verificar_autenticidad("gokulegana@hotmail_gmail_twitter.com", "1234 5678 9012 3456")
print("Cliente autenticado:", "Si" if autenticado else "No")

File: Sesion 7/Actividades/Actividad 1/actividad_1.java
import java.util.*;

class Cliente {
    protected String nombre;
    protected String apellido;
    protected String correo;
    protected String direccion;
    protected String numero_tarjeta;

    public Cliente(String nombre, String apellido, String correo, String direccion, String numero_tarjeta) {
        this.nombre = nombre;
        this.apellido = apellido;
        this.correo = correo;
        this.direccion = direccion;
        this.numero_tarjeta = numero_tarjeta;
    }

    public String obtenerInformacion() {
        String info = "";
        info += "Nombre: " + nombre + "\n";
        info += "Apellido: " + apellido + "\n";
        info += "Correo: " + correo + "\n";
        info += "Direccion: " + direccion + "\n";
        info += "Numero de tarjeta: " + numero_tarjeta + "\n";
        return info;
    }
}

class ClienteSeguro extends Cliente {
    public ClienteSeguro(String nombre, String apellido, String correo, String direccion, String numero_tarjeta) {
        super(nombre, apellido, correo, direccion, numero_tarjeta);
    }

    public String encriptarInformacion() {
        String encriptacion = "";
        for (char c : correo.toCharArray()) { // Itera cada letra de correo
            encriptacion += Integer.toString((int) c); // Convierte cada letra en ASCII
        }
        return encriptacion;
    }

    public boolean verificarAutenticidad(String correo_ingresado, String numero_tarjeta_ingresado) {
        return correo_ingresado.equals(correo) && numero_tarjeta_ingresado.equals(numero_tarjeta);
    }
}

public class Main {
    public static void main(String[] args) {
        ClienteSeguro clienteSeguro1 = new ClienteSeguro("goku", "sayayin 1000", "gokulegana@hotmail_gmail_twitter.com", "Bajo un Puente", "1234 5678 9012 3456");

        System.out.println("Informacion del cliente:");
        System.out.println(clienteSeguro1.obtenerInformacion());

        System.out.println("Informacion encriptada: " + clienteSeguro1.encriptarInformacion());

        boolean autenticado = clienteSeguro1.verificarAutenticidad("gokulegana@hotmail_gmail_twitter.com", "1234 5678 9012 3456");
        System.out.println("Cliente autenticado: " + (autenticado ? "Si" : "No"));
    }
}
